### PR TITLE
Add smoke test to lighty-community-aaa-restconf-app

### PR DESCRIPTION
**Pull request by PeterSuna**

### CODE REVIEW

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package io.lighty.kit.examples.community;
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:1]: Add license header

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import io.lighty.kit.examples.community.aaa.restconf.Main;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import org.eclipse.jetty.client.HttpClient;
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:8]: Use native java HTTP library

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public void startUp() throws Exception {
+        Main.start();
+
+        this.httpClient = new HttpClient();
+        this.httpClient.setIdleTimeout(60000 * 60);
+        this.httpClient.start();
+    }
+
+    @Test
+    public void readDataCorrectCredentials() throws Exception {
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:36]: Add at least one negative test with wrong credentials

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        Assert.assertEquals(sendGetRequestJSON(this.httpClient, TEST_ADDRESS).getStatus(), HttpStatus.OK_200);
+    }
+
+    private ContentResponse sendGetRequestJSON(final HttpClient client, final String path)
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:40]: Use some meaningful timeout for REST request so build/test will not hang too long if something goes wrong

**Review by Pyro2266 — CHANGES_REQUESTED** [review]

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        throws InterruptedException, ExecutionException, TimeoutException {
+        return client.newRequest(path)
+            .method(HttpMethod.GET)
+            .header(AUTH, BASIC_AUTH)
+            .send();
+    }
+
+    @AfterClass
+    public void tearDown() throws Exception {
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:49]: Shut down lighty app too (you will need to add shutdown method to app itself)

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java (hunk 1 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:9]: Done

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java (hunk 2 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java]: Done

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java (hunk 3 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java:50]: Done

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java (hunk 4 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java]: Done

File: lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java (hunk 5 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/test/java/io/lighty/kit/examples/community/CommunityAAARestconfAppTest.java]: Done

File: lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java [hunk 6]
```diff
@@ -47,9 +47,10 @@
     private static final Logger LOG = LoggerFactory.getLogger(Main.class);
     private static final String PASS = "bar";
     private static final String USER = "foo";
+    private static ShutdownHook shutdownHook;
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java:50]: ShutdownHook should not be static

File: lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java [hunk 7]
```diff
@@ -108,15 +109,20 @@ private static void startLighty(final ControllerConfiguration controllerConfigur
...
 
-    public static void start() throws Exception {
-        main(new String[]{});
+    public void shutdown() {
+        shutdownHook.execute();
     }
 
-    private Main() {
-        /** Hide constructor, this is a utility class **/
+    public static void main(String[] args) {
```
Comment (Pyro2266) [inline lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java:123]: Move main method to the top of this class

**Review by Pyro2266 — CHANGES_REQUESTED** [review]

File: lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java (hunk 6 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java]: done

File: lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java (hunk 7 above)
Comment (PeterSuna) [reply lighty-examples/lighty-community-aaa-restconf-app/src/main/java/io/lighty/kit/examples/community/aaa/restconf/Main.java]: done

**Pull request merged**